recipes/pcre2/all/conanfile.py

```diff
@@ -90,26 +105,46 @@ def package(self):
         tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
 
     def package_info(self):
-        def library_name(library):
-            if self.settings.build_type == "Debug" and self.settings.os == "Windows":
-                library += "d"
-            if self.settings.compiler == "gcc" and self.settings.os == "Windows" and self.options.shared:
-                library += ".dll"
-            return library
-
-        # May need components for
-        # ./lib/pkgconfig/libpcre2-32.pc
-        # ./lib/pkgconfig/libpcre2-8.pc
-        # ./lib/pkgconfig/libpcre2-posix.pc
-        # ./lib/pkgconfig/libpcre2-16.pc
-
         self.cpp_info.names["pkg_config"] = "libpcre2"
-        self.cpp_info.libs = [library_name("pcre2-posix")]
         if self.options.build_pcre2_8:
-            self.cpp_info.libs.append(library_name("pcre2-8"))
+            # pcre2-8
+            self.cpp_info.components["pcre2-8"].names["pkg_config"] = "libpcre2-8"
+            self.cpp_info.components["pcre2-8"].libs = [self._lib_name("pcre2-8")]
+            if not self.options.shared:
+                self.cpp_info.components["pcre2-8"].defines.append("PCRE2_STATIC")
+            # pcre2-posix
+            self.cpp_info.components["pcre2-posix"].names["pkg_config"] = "libpcre2-posix"
+            self.cpp_info.components["pcre2-posix"].libs = [self._lib_name("pcre2-posix")]
+            self.cpp_info.components["pcre2-posix"].requires = ["pcre2-8"]
+        # pcre2-16
         if self.options.build_pcre2_16:
-            self.cpp_info.libs.append(library_name("pcre2-16"))
+            self.cpp_info.components["pcre2-16"].names["pkg_config"] = "libpcre2-16"
+            self.cpp_info.components["pcre2-16"].libs = [self._lib_name("pcre2-16")]
+            if not self.options.shared:
+                self.cpp_info.components["pcre2-16"].defines.append("PCRE2_STATIC")
+        # pcre2-32
         if self.options.build_pcre2_32:
-            self.cpp_info.libs.append(library_name("pcre2-32"))
-        if not self.options.shared:
-            self.cpp_info.defines.append("PCRE2_STATIC")
+            self.cpp_info.components["pcre2-32"].names["pkg_config"] = "libpcre2-32"
+            self.cpp_info.components["pcre2-32"].libs = [self._lib_name("pcre2-32")]
+            if not self.options.shared:
+                self.cpp_info.components["pcre2-32"].defines.append("PCRE2_STATIC")
+
+        if self.options.build_pcre2grep:
+            bin_path = os.path.join(self.package_folder, "bin")
+            self.output.info("Appending PATH environment variable: {}".format(bin_path))
+            self.env_info.PATH.append(bin_path)
+            # FIXME: This is a workaround to avoid ConanException. zlib and bzip2
```

Comment:
could there be a component for pcre2grep?

Reply:
I don't think that conan generators support executable targets.

Reply:
please submit an issue, if needed